project_task_scheduling/wizards/scheduling_wizard.py

```diff
@@ -0,0 +1,397 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from collections import namedtuple
+from datetime import datetime, time, timedelta
+from functools import cmp_to_key
+import logging
+import math
+import random
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.float_utils import float_compare
+from odoo.tools.translate import _
+
+from odoo.addons.resource.models import resource as resource_model
+
+_logger = logging.getLogger(__name__)
+
+
+class ProjectTaskSchedulingWizard(models.TransientModel):
+    _name = "project.task.scheduling.wizard"
+    _description = "Project task scheduling wizard"
+
+    _state_obj = namedtuple('State', (
```

Comment:
Could you separate the scheduling algorithm from the wizard?
An idea could be to move the `_state_obj` structure and the related algorithm methods to a separate abstract model which represents the scheduling algorithm. You could then make the wizard inheriting from that model.

This way you will make the module more easy to be customized or extended.
What do you think?

Reply:
It seems good to me, thank you for the idea, i think it's good, I already have plans to work on this, but I have to organize the ideas.